server/sonar-web/src/main/js/api/issues.ts

```diff
@@ -45,7 +48,16 @@ export function searchIssues(query: RequestData): Promise<IssuesResponse> {
   return getJSON('/api/issues/search', query);
 }
 
-export function getFacets(query: RequestData, facets: string[]): Promise<any> {
+export function getFacets(
+  query: RequestData,
+  facets: string[]
+): Promise<{
+  facets: Array<{
```

Comment:
I think `Array<>` is more readable when you have a complex (usually multiline) definition, otherwise `[]` is good.

I think I more or less follow the [`array-type`](https://palantir.github.io/tslint/rules/array-type/) tslint rule with the `"array-simple"` option.

Reply:
Ok lets activate this rule with a warning so we enforce it then :)

Reply:
let's install tslint first 😉